src/connection.js

```diff
@@ -591,43 +615,106 @@ export class Connection {
   /**
    * Fetch the balance for the specified public key
    */
-  async getBalance(
+  async getBalanceAndContext(
     publicKey: PublicKey,
     commitment: ?Commitment,
-  ): Promise<number> {
+  ): Promise<RpcResponseAndContext<number>> {
     const args = this._argsWithCommitment([publicKey.toBase58()], commitment);
     const unsafeRes = await this._rpcRequest('getBalance', args);
-    const res = GetBalanceRpcResult(unsafeRes);
+    const res = GetBalanceAndContextRpcResult(unsafeRes);
     if (res.error) {
       throw new Error(res.error.message);
     }
     assert(typeof res.result !== 'undefined');
-    return res.result;
+
+    const isV21 =
+      typeof res.result.context !== 'undefined' &&
+      typeof res.result.value !== 'undefined';
+
+    const slot = isV21 ? res.result.context.slot : NaN;
```

Comment:
This might be a little simpler as:
```
if (isV21) {
  return res.result;
} else {
  return {
    context: {
      NaN,
    },
    res.result,
  };
}
```
...assuming flow is okay with that. (I've been deep in Rustland for a while)

Reply:
ooh nice idea!